test/functional/tests/io_class/test_io_class_directory.py

```diff
@@ -173,6 +174,7 @@ def test_ioclass_directory_file_operations(filesystem):
         Directory.create_directory(path=nested_dir_path, parents=True)
         sync()
         drop_caches(DropCachesMode.ALL)
+        time.sleep(3)
```

Comment:
sleeping an arbitrary amount of time is frowned upon

Reply:
So what do you suggest? There is no possibility that a event which needs to be finished before continuing the test can be tracked from within the test.
The root cause of the issue is that the directory based classifier does not check if a directory exists in a continuous manner. It scans the filesystem every 1 second. If some IO occurs to the newly created directory before the directory has been detected, it will not be identified as targeted to that ioclass. Allowing the classifier to settle resolves the issue. 

Nevertheless, if you have any idea for better workaround, I will be eager to implement it.

Reply:
If it's a known value (the scan every second) - I'd put it in a variable with a descriptive name. We do have a variable in ioclass_config.py that determines maximum time for reclassification (6s). Would it be the variable we need here?